### PR TITLE
util/tracing, sql: expose span's goroutine IDs

### LINKED ISSUE #59200: tracing: track goroutine ID in Span
**Is your feature request related to a problem? Please describe.**
While `Span` can cross goroutine boundaries, idiomatically it should not. This gives us a straightforward way to, in many cases, heuristically find the goroutine associated to a long-running trace span. Note that the tracked goroutine may already have terminated, and the goroutine ID may now be reused by a different goroutine. This is why it says "heuristically" - in practice this should work well.

**Describe the solution you'd like**

We grab the goroutine ID (`goid` package does it) at span creation time (we could also update it on every access, but not sure this buys us much) and can expose it as part of https://github.com/cockroachdb/cockroach/issues/55733.

**Describe alternatives you've considered**

https://github.com/cockroachdb/cockroach/issues/58609 suggests allowing to hook into the output of a span (making it verbose if need be). However, that gives no clue as to the location of a truly stuck span (say a goroutine sitting on a `select`). It also involves more complexity.

**Additional context**

We are at the moment targeting "expert debugging" for this use case only. Chasing down a goroutine in this way will be too complicated for end users, but should be within the realm of engineers, TSEs, CC-SREs, CS, etc with proper playbooks.

### PR DESCRIPTION
**Pull request by angelapwen**
Resolves #59200. 

This commit adds a goroutine ID to each span's recording, as well
as as a column in the `crdb_internal.node_inflight_trace_spans`
virtual table. The goroutine ID is retrieved upon span creation.

Release note (sql change): adds `goroutine_id` column to the
`crdb_internal.node_inflight_trace_spans` virtual table that
represents the goroutine ID associated with a particular span.

### CODE REVIEW

File: pkg/util/tracing/crdbspan.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ type crdbSpan struct {
 	traceID      uint64 // probabilistically unique.
 	spanID       uint64 // probabilistically unique.
 	parentSpanID uint64
+	goroutineID  uint64 // goroutine ID associated with the span.
```
Comment (irfansharif) [inline pkg/util/tracing/crdbspan.go:35]: [nit] We can drop the comment, pretty self explanatory me thinks.

**Review by irfansharif — APPROVED** [review]

File: pkg/util/tracing/tracingpb/recorded_span.proto [hunk 2]
```diff
@@ -76,6 +76,9 @@ message RecordedSpan {
   //
   // TODO(tbg): rename once DeprecatedStats is removed.
   repeated google.protobuf.Any internal_structured = 11;
+
+  // Goroutine ID associated with this span at time of span creation.
```
Comment (irfansharif) [inline pkg/util/tracing/tracingpb/recorded_span.proto:80]: [nit] Here, and above, I'd phrase it as "ID of the goroutine creating the span."

**Comment by angelapwen** [comment]
Thanks for the review @irfansharif 💝

**Review by tbg — APPROVED** [review]
Looks good!

> Reviewed 4 of 8 files at r1, 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59717#-:-MSbcQVm2L8xUYb-hr3s:b-vs1prh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @angelapwen, @irfansharif, and @knz)

___
*[pkg/util/tracing/tracingpb/recorded_span.proto, line 80 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59717#-MSZIxJSVVCmmf6xbZST-r1-80:-MSbbtC00ORxMb3Mm854:b-qmjqft) ([raw file](https://github.com/cockroachdb/cockroach/blob/baa91dcf235475bb28656769469ad27a35f3f208/pkg/util/tracing/tracingpb/recorded_span.proto#L80)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

[nit] Here, and above, I'd phrase it as "ID of the goroutine creating the span."
</blockquote></details>

Or, more precisely, "The ID of the goroutine on which the span was created."

**Review by knz — APPROVED** [review]

**Review by angelapwen — COMMENTED** [review]
bors r=irfansharif 

Thank you everyone for the review! 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59717#-:-MScAq37EI0Ffkj2d7U3:b8p8lhs)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @angelapwen, @irfansharif, and @tbg)

___
*[pkg/util/tracing/tracingpb/recorded_span.proto, line 80 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59717#-MSZIxJSVVCmmf6xbZST-r1-80:-MScAhpQ4kaRNefoSTbn:b-7sh45b) ([raw file](https://github.com/cockroachdb/cockroach/blob/baa91dcf235475bb28656769469ad27a35f3f208/pkg/util/tracing/tracingpb/recorded_span.proto#L80)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Or, more precisely, "The ID of the goroutine on which the span was created."
</blockquote></details>

Done! Thanks all for the precise language :smile:

**Pull request merged**